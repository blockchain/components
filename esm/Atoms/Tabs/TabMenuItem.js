function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  position: absolute;\n  bottom: -2px;\n  left: 0;\n  width: 100%;\n  height: 5px;\n  transition-duration: 0.3s;\n  transition-timing-function: ease-in-out;\n  background-color: ", ";\n  opacity: ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  position: relative;\n  flex: 1 auto;\n  display: flex;\n  flex-direction: row;\n  justify-content: center;\n  align-items: center;\n  padding: 15px;\n  box-sizing: border-box;\n  cursor: pointer;\n\n  &:hover {\n    & :last-child {\n      opacity: 1 !important;\n    }\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
var Wrapper = styled.div(_templateObject());
var Bar = styled.div(_templateObject2(), function (props) {
  return props.theme['gold'];
}, function (props) {
  return props.visible ? '1' : '0';
});

var TabMenuItem = function TabMenuItem(_ref) {
  var children = _ref.children,
      selected = _ref.selected,
      rest = _objectWithoutProperties(_ref, ["children", "selected"]);

  return React.createElement(Wrapper, rest, children, React.createElement(Bar, {
    visible: selected
  }));
};

TabMenuItem.propTypes = {
  children: PropTypes.node,
  selected: PropTypes.bool
};
TabMenuItem.defaultProps = {
  selected: false
};
export default TabMenuItem;