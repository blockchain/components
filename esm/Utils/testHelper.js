import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}