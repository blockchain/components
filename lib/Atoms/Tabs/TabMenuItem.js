"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  position: absolute;\n  bottom: -2px;\n  left: 0;\n  width: 100%;\n  height: 5px;\n  transition-duration: 0.3s;\n  transition-timing-function: ease-in-out;\n  background-color: ", ";\n  opacity: ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  position: relative;\n  flex: 1 auto;\n  display: flex;\n  flex-direction: row;\n  justify-content: center;\n  align-items: center;\n  padding: 15px;\n  box-sizing: border-box;\n  cursor: pointer;\n\n  &:hover {\n    & :last-child {\n      opacity: 1 !important;\n    }\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Wrapper = _styledComponents.default.div(_templateObject());

var Bar = _styledComponents.default.div(_templateObject2(), function (props) {
  return props.theme['gold'];
}, function (props) {
  return props.visible ? '1' : '0';
});

var TabMenuItem = function TabMenuItem(_ref) {
  var children = _ref.children,
      selected = _ref.selected,
      rest = _objectWithoutProperties(_ref, ["children", "selected"]);

  return _react.default.createElement(Wrapper, rest, children, _react.default.createElement(Bar, {
    visible: selected
  }));
};

TabMenuItem.propTypes = {
  children: _propTypes.default.node,
  selected: _propTypes.default.bool
};
TabMenuItem.defaultProps = {
  selected: false
};
var _default = TabMenuItem;
exports.default = _default;