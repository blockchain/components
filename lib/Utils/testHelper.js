"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}