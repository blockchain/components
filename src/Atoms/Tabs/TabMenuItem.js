// @flow strict
import * as React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import injectE2E from '../../Tools/injectE2E'

const Wrapper = styled.div.attrs({
  'data-e2e': injectE2E,
})`
  position: relative;
  flex: 1 auto;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  padding: 1rem;
  box-sizing: border-box;
  cursor: pointer;

  @media (max-width: ${(props) => props.theme.screenSizes.sm}) {
    order: 1;

    &.hidden {
      order: 2;
    }
  }

  @media (min-width: ${(props) => props.theme.screenSizes.sm}) {
    display: flex;
    justify-content: center;
    order: 1;

    &:hover {
      & :last-child {
        opacity: 1;
      }
    }
  }
`
const Bar = styled.div`
  position: absolute;
  bottom: -2px;
  left: 0;
  width: 100%;
  height: 0.2rem;
  transition-duration: 0.3s;
  transition-timing-function: ease-in-out;
  background-color: ${(props) => props.theme['tabMenuBarColor']};
  opacity: ${(props) => (props.selected ? '1' : '0')};
`

type PropsType = {
  +children?: React.Node,
  +className?: string,
  +e2e?: string,
  +selected?: boolean,
}

const TabMenuItem = ({ children, className, selected, ...rest }: PropsType) => (
  <Wrapper
    className={selected === true ? className : `${className || ''} disabled`}
    {...rest}
  >
    {children}
    <Bar selected={selected} />
  </Wrapper>
)

TabMenuItem.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  e2e: PropTypes.string,
  selected: PropTypes.bool,
}

TabMenuItem.defaultProps = {
  e2e: 'defaultTabMenuItem',
  selected: false,
}

export default TabMenuItem
