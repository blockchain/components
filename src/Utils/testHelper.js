import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
