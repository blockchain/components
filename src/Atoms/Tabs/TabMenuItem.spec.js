import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import TabMenuItem from './TabMenuItem'

describe('TabMenuItem', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<TabMenuItem>MyTabMenuItem</TabMenuItem>)

    expect(tree.find(TabMenuItem)).toMatchSnapshot()
  })

  it('should render correctly when selected', () => {
    const tree = mountWithTheme(
      <TabMenuItem selected>MyTabMenuItem</TabMenuItem>,
    )

    expect(tree.find(TabMenuItem)).toMatchSnapshot()
  })
})
