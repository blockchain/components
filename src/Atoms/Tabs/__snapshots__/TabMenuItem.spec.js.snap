// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabMenuItem should render correctly 1`] = `
.c0 {
  position: relative;
  -webkit-flex: 1 auto;
  -ms-flex: 1 auto;
  flex: 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem;
  box-sizing: border-box;
  cursor: pointer;
}

.c1 {
  position: absolute;
  bottom: -2px;
  left: 0;
  width: 100%;
  height: 0.325rem;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  background-color: #FFCA00;
  opacity: 0;
}

@media (max-width:48rem) {
  .c0 {
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
  }

  .c0.hidden {
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }
}

@media (min-width:48rem) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
  }

  .c0:hover :last-child {
    opacity: 1;
  }
}

<TabMenuItem
  selected={false}
>
  <styled.div
    className="undefined disabled"
  >
    <div
      className="undefined disabled c0"
    >
      MyTabMenuItem
      <styled.div
        selected={false}
      >
        <div
          className="c1"
          selected={false}
        />
      </styled.div>
    </div>
  </styled.div>
</TabMenuItem>
`;

exports[`TabMenuItem should render correctly when selected 1`] = `
.c0 {
  position: relative;
  -webkit-flex: 1 auto;
  -ms-flex: 1 auto;
  flex: 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem;
  box-sizing: border-box;
  cursor: pointer;
}

.c1 {
  position: absolute;
  bottom: -2px;
  left: 0;
  width: 100%;
  height: 0.325rem;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  background-color: #FFCA00;
  opacity: 1;
}

@media (max-width:48rem) {
  .c0 {
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
  }

  .c0.hidden {
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }
}

@media (min-width:48rem) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
  }

  .c0:hover :last-child {
    opacity: 1;
  }
}

<TabMenuItem
  selected={true}
>
  <styled.div>
    <div
      className="c0"
    >
      MyTabMenuItem
      <styled.div
        selected={true}
      >
        <div
          className="c1"
          selected={true}
        />
      </styled.div>
    </div>
  </styled.div>
</TabMenuItem>
`;
